src/L8/game/checkers/checkers_utils.py

```diff
@@ -0,0 +1,117 @@
+from L8.board.checkers_board import CheckersBoard
+from L8.game.checkers.checkers_constans import CheckerColors
+from L8.game.checkers.checkers_move import CheckersMove
+
+
+class CheckerGameUtil:
+
+    @staticmethod
+    def is_valid_move(move: CheckersMove, player_color: str, board: CheckersBoard) -> bool:
+        """
+        Checks if move is a legal move by obtaining a list of legal moves and checking if the move is in the list.
+        :param move:
+        :param player_color:
+        :param board:
+        :return: True if it is legal, false otherwise
+        """
+        legal_moves = CheckerGameUtil.get_valid_moves_for_player(board, player_color)
```

Comment:
What's the cost of calculating the whole solution space (even for moves that the player is not currently making)? Wouldn't it be better if we calculate if the current move is valid?